tests/functional/test_lambda.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) 2025, Crash Override, Inc.
+#
+# This file is part of Chalk
+# (see https://crashoverride.com/docs/chalk)
+from pathlib import Path
+import shutil
+import zipfile
+
+import pytest
+
+from .chalk.runner import Chalk
+from .conf import ZIPS
+from .utils.log import get_logger
+
+
+logger = get_logger()
+
+
+# Test data fixtures
+PYTHON_LAMBDA_ZIP = ZIPS / "python" / "my_deployment_package.zip"
+NODEJS_LAMBDA_ZIP = ZIPS / "nodejs" / "function.zip"
+GOLANG_LAMBDA_ZIP = ZIPS / "golang" / "myFunction.zip"
+MISC_LAMBDA_ZIP = ZIPS / "misc" / "misc.zip"
+EMPTY_ZIP = ZIPS / "empty" / "empty.zip"
+
+
+def verify_chalk_binary_in_zip(zip_path: Path, extract_dir: Path):
+    """Helper function to verify chalk binary exists in the zip file."""
+    # Extract the zip contents
+    extract_dir.mkdir(exist_ok=True)
+
+    with zipfile.ZipFile(zip_path, "r") as zip_ref:
+        zip_ref.extractall(extract_dir)
+
+    # Verify the chalk binary was added
+    chalk_binary_path = extract_dir / "chalk"
+    assert chalk_binary_path.exists(), "Chalk binary was not found in the zip"
```

Comment:
in pytest adding explicit assertion message is not necessary as it prints useful context about the error:

given:

```python
def a():
    return 1
def b():
    return 2
def test_foo():
    assert a() == b()
```

error will be:

```python
    def test_foo():
>       assert a() == b()
E       assert 1 == 2
E        +  where 1 = a()
E        +  and   2 = b()
```